apps/etterlatte-vedtaksvurdering/src/main/kotlin/vedtaksvurdering/VedtaksvurderingRepository.kt

```diff
@@ -56,46 +57,75 @@ class VedtaksvurderingRepository(val datasource: DataSource) {
                         "beregningsresultat" to nyttVedtak.beregning?.toJson(),
                         "vilkaarsresultat" to nyttVedtak.vilkaarsvurdering?.toJson()
                     )
-                ).let { query -> tx.run(query.asUpdateAndReturnGeneratedKey) }
-
-                nyttVedtak.utbetalingsperioder.forEach {
-                    queryOf(
-                        statement = """
-                            INSERT INTO utbetalingsperiode(vedtakid, datofom, datotom, type, beloep) 
-                            VALUES (:vedtakid, :datofom, :datotom, :type, :beloep)
-                            """,
-                        mapOf(
-                            "vedtakid" to vedtakId,
-                            "datofom" to it.periode.fom.atDay(1).let(Date::valueOf),
-                            "datotom" to it.periode.tom?.atEndOfMonth()?.let(Date::valueOf),
-                            "type" to it.type.name,
-                            "beloep" to it.beloep
-                        )
-                    ).let { query -> tx.run(query.asUpdate) }
-                }
+                )
+                    .let { query -> tx.run(query.asUpdateAndReturnGeneratedKey) }
+                    ?.let { vedtakId ->
+                        opprettUtbetalingsperioder(vedtakId, nyttVedtak.utbetalingsperioder, tx)
+                    } ?: throw Exception("Kunne ikke opprette vedtak for behandling ${nyttVedtak.behandlingId}")
             }.let {
-                hentVedtak(nyttVedtak.behandlingId) ?: throw Exception("Kunne ikke opprette vedtak")
+                hentVedtak(nyttVedtak.behandlingId)
+                    ?: throw Exception("Kunne ikke opprette vedtak for behandling ${nyttVedtak.behandlingId}")
             }
         }
-    }
 
-    fun oppdaterVedtak(vedtak: Vedtak): Vedtak = repositoryWrapper.oppdater(
-        query = """
-            UPDATE vedtak 
-            SET datovirkfom = :datovirkfom, beregningsresultat = :beregningsresultat, 
-                vilkaarsresultat = :vilkaarsresultat 
-            WHERE behandlingId = :behandlingid
-        """,
-        params = mapOf(
-            "datovirkfom" to vedtak.virkningstidspunkt.atDay(1),
-            "beregningsresultat" to vedtak.beregning?.toJson(),
-            "vilkaarsresultat" to vedtak.vilkaarsvurdering?.toJson(),
-            "behandlingid" to vedtak.behandlingId
-        ),
-        loggtekst = "Oppdaterer vedtak behandlingid: ${vedtak.behandlingId}"
-    )
-        .also { require(it == 1) }
-        .let { hentVedtak(vedtak.behandlingId) ?: throw Exception("Kunne ikke opprette vedtak") }
+    fun oppdaterVedtak(oppdatertVedtak: Vedtak): Vedtak =
```

Comment:
Skal vera forsiktig med å foreslå at denne PR-en skal bli større, men her trur eg vi kunne forenkla ein del ved å bruke `KotliqueryRepositoryWrapper`

Reply:
Så bare på opprett at det ikke var støtte for flere operasjoner i en felles transaksjon - men ser nå at det finnes et `ekstra` parameter i oppdater-metoden der. Kunne det vært mulig og også hente ut generert id på noe vis med den wrapperen?

Reply:
Ja, det vil eg tru. Både `oppdater` og `opprettFlere` returnerer Int/List<Int> i dag, så det bør ikkje vera store endringa som skal til.

Elles er jo den klassa relativt ny, så trur det er berre fint å lage støtte for fleire typar greier vi gjer mot databasen der.

Reply:
Gjør dette i en egen sak evnt